### ISSUE TITLE
VLAN Interface Errors

### ISSUE THREAD

**Issue by tomatotoast** [opnsense/core#4235]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md

[X] I have searched the existing issues and I'm convinced that mine is new.

**Describe the bug**
A clear and concise description of what the bug is, including last known working version (if any).

I just upgraded a box from 20.1 to 20.7 Since the upgrade I can detect errors on the VLANs. On the side of the switch I could not detect any errors.

*Tip*: to validate your setup was working with the previous version, use opnsense-revert (https://docs.opnsense.org/manual/opnsense_tools.html#opnsense-revert)

**To Reproduce**
1. Use box with igb NICs and 20.1
2. Create VLAN on Parent Interface and create some Traffic, All interface Errors should be "0"
3. Upgrade to 20.7
4. Look at Interface Errors

**Expected behavior**
Like in 20.1 in 20.7 there should be no Errors on the VLAN Interfaces

**Screenshots**
![grafik](https://user-images.githubusercontent.com/60579201/89105588-94e16f00-d422-11ea-983b-6ed98e7dbfba.png)
![grafik](https://user-images.githubusercontent.com/60579201/89105593-99a62300-d422-11ea-85f6-f28225b0994d.png)

**Relevant log files**
If applicable, information from log files supporting your claim.

**Additional context**

**Environment**
Software version used and hardware type if relevant.
e.g.:

OPNsense 20.7-amd64
FreeBSD 12.1-RELEASE-p7-HBSD
OpenSSL 1.1.1g 21 Apr 2020

    vendor     = 'Intel Corporation'
    device     = 'I211 Gigabit Network Connection'
    class      = network
    subclass   = ethernet

**Comment by tomatotoast** [comment]
In case someone asks:
Suricata is not in use.

**Comment by mimugmail** [comment]
Can you try this on Vanilla FreeBSD so we know if it's upstream or not?

**Comment by tomatotoast** [comment]
I think this is not possible as the machine is running in production.

**Comment by L1ghtn1ng** [comment]
@tomatotoast can you try and see what happens with vlan hardware filtering with it being disabled?

**Comment by marjohn56** [comment]
Just completed the upgrade to 20.7, I'm now also seeing these errors. This was an upgrade from a dev path that was on FreeBSD 11.2 so I think this could be a FreeBSD 12 issue.

**Comment by AdSchellevis** [comment]
Question is if the same happens to cards as well, on one of our boxes I also see a (very low) number of errors on the vlan interfaces. I didn't notice any actual disruptions while testing so it might also be a glitch happening when removing vlan hardware offloading or some other hardware interface tag.

**Comment by marjohn56** [comment]
Yes, mine to, 6, 12, and 13 across the three VLANs. Would like a way to clear the count, is there?

**Comment by marjohn56** [comment]
Ah, just renamed one of the VLANs, which of course caused a reset and now I have a count of 7, 28 & 25... doesn't appear to increase beyond that initial count when the interface(s) are reset.

**Comment by peter-vanpoucke** [comment]
Not sure if this is related, but next to the interface errors, I notice traffic gets duplicated in the VLAN. If a ping is send between devices on the same VLAN, OPNsense seems to intercept and resend the ping request and the replies.

When reverting to the snapshot with OPNsense 20.1.9_1-amd64 the issue no longer occurs. Running on a VMWare ESXi 6.7.0 Update 3.

**Comment by peter-vanpoucke** [comment]
> Not sure if this is related, but next to the interface errors, I notice traffic gets duplicated in the VLAN. If a ping is send between devices on the same VLAN, OPNsense seems to intercept and resend the ping request and the replies.
> 
> When reverting to the snapshot with OPNsense 20.1.9_1-amd64 the issue no longer occurs. Running on a VMWare ESXi 6.7.0 Update 3.

Issue with duplicated packages on VLAN interfaces still occurs in 20.7.1.

**Comment by mimugmail** [comment]
This is an issue of VMware and promisc mode

**Comment by fichtner** [comment]
For Intel network drivers in the guest can you try this kernel?

    # opnsense-update -rk 20.7.1-nopromisc
    #  opnsense-shell reboot

**Comment by fichtner** [comment]
Should be fixed in 20.7.2. If you guys could confirm that would be great.
